datafusion/core/tests/dataframe/dataframe_functions.rs

```diff
@@ -346,7 +346,7 @@ async fn test_fn_approx_median() -> Result<()> {
         "+-----------------------+",
         "| approx_median(test.b) |",
         "+-----------------------+",
-        "| 10                    |",
+        "| 10.0                  |",
```

Comment:
this seems like a change in behavior -- with this PR  now median always returns float but before it returned the same type as its input

Reply:
Yes, I change the result to f64 now.

I think it is fine to have f64 for median value. I check the result of Duckdb, they have double for integer, although they have decimal for decimal input, but since we doesn't support decimal for approx_median so there is no regression.